### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by muskan2532150**
Add the following validations:
 
- For the User model:
     - Name must not be blank.
     - PostsCounter must be an integer greater than or equal to zero.
 - For the Post model:
     - Title must not be blank.
     - Title must not exceed 250 characters.
     - CommentsCounter must be an integer greater than or equal to zero.
     - LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
...
+  validates_numericality_of :likes_counter, only_integer: true, presence: true, greater_than_or_equal_to: 0
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
+
+  def five_recent_comments
+    Comment.where(post_id: id).last(5)
+  end
 end
```
Comment (fobadara) [inline app/models/post.rb]: [REQUIRED] You should create tests for custom methods including update_posts_count and five_recent_commnets.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,12 @@
...
+  has_many :likes, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :posts, foreign_key: 'author_id'
+
+  validates :name, presence: true, allow_blank: false
+  validates_numericality_of :posts_counter, only_integer: true, presence: true, greater_than_or_equal_to: 0
+
+  def three_recent_posts
+    posts.where(author_id: id).last(3)
+  end
```
Comment (fobadara) [inline app/models/user.rb:12]: [REQUIRED] You should create tests for custom methods including three_recent_posts

File: app/models/like.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
 class Like < ApplicationRecord
+  after_save :update_likes_count
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def update_likes_count
+    post.increment!(:likes_counter)
+  end
 end
```
Comment (fobadara) [inline app/models/like.rb]: [REQUIRED] I can't seem to find specs for like model

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,2 +1,11 @@
...
+  after_save :update_comments_count
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  validates :text, length: { maxmium: 250 }, presence: true
+
+  def update_comments_count
+    post.increment!(:comments_counter)
+  end
 end
```
Comment (fobadara) [inline app/models/comment.rb]: [REQUIRED] I can't seem to find specs for comment

**Review by fobadara — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️ 

Hi @muskan2532150, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Project looks profesional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (fobadara) [inline app/models/user.rb:12]: [REQUIRED] You didn't implement the changes I suggested. Kindly create a test for this method

File: app/models/post.rb (hunk 1 above)
Comment (fobadara) [inline app/models/post.rb]: [REQUIRED] You didn't implement the changes I suggested. Kindly create a test for these methods

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @muskan2532150, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
✅ Linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 5]
```diff
@@ -1,24 +1,157 @@
...
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- SUPPORT -->
+
+## ⭐️ Show your support <a name="support"></a>
+
```
Comment (1GORDON) [inline README.md:138]: Great job @muskan2532150 🤝

- [ ] [Optional] Kindly fix the broken links in your readme file to make it more professional. You may fix the `author`, `issues page`, and the `MIT license` links.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @muskan2532150 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️

## Optional suggestions

- [ ] You may always write your commit messages in an imperative tense something like `Fix Linter Errors` instead of `Fixed Linter Errors`.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**